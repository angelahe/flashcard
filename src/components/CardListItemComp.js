import React from 'react';
import PropTypes from 'prop-types';
import '../styles/flashcard.css';
import editbtn from '../img/buttons/edit_FFFFFF.png';
import deletebtn from '../img/buttons/delete_FFFFFF.png';

const CardListItemComp = (props) => {
  const {
    card,
    onCardSelect,
    onCardEdit,
    onCardDelete,
  } = props;

  function handleItemClick() {
    onCardSelect(card);
  }

  function handleEditClick() {
    onCardEdit(card);
  }

  function handleDeleteClick() {
    onCardDelete(card.card_id);
  }

  return (
    <div
      className="ItemBox AppListItem CardListItemComp"
      role="button"
      tabIndex="0"
    >
      <span
        className="CardListItemText DetailText"
        role="button"
        tabIndex="0"
        onClick={handleItemClick}
        onKeyPress={handleItemClick}
      >{card.L1_word}
      </span>
      <button className="CardListItemEdit AppBtn" type="button" onClick={handleEditClick}>
        <img className="btnImg" src={editbtn} alt="Add" />
      </button>
      <button className="CardListItemDelete AppBtn Delete" type="button" onClick={handleDeleteClick}>
        <img className="btnImg" src={deletebtn} alt="Delete" />
      </button>
      <hr />
    </div>
  );
};

CardListItemComp.defaultProps = {
  onCardSelect: () => { },
  onCardEdit: () => { },
  onCardDelete: () => { },
};

CardListItemComp.propTypes = {
  card: PropTypes.shape({
    card_id: PropTypes.string.isRequired,
  }).isRequired,
  onCardSelect: () => { },
  onCardEdit: () => { },
  onCardDelete: () => { },
};
export default CardListItemComp;
