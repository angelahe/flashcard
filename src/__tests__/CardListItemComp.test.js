import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import CardListItemComp from '../components/CardListItemComp';

const card = { card_id: 'ABCD-1234' };

it('renders without crashing', () => {
  shallow(<CardListItemComp card={card} />);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CardListItemComp card={card} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('snapshot of UI renders consistently', () => {
  const tree = renderer.create(<CardListItemComp card={card} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the container div CardListItemComp on shallow render', () => {
  const component = shallow(<CardListItemComp card={card} />);
  expect(component.html()).toContain('CardListItemComp');
  expect(component.html()).toContain('CardListItemText');
  expect(component.html()).toContain('CardListItemEdit');
  expect(component.html()).toContain('CardListItemDelete');
});

it('click editCard button calls handler', (done) => {
  const handleCardEdit = jest.fn(() => {});
  const component = shallow(<CardListItemComp card={card} onCardEdit={handleCardEdit} />);
  const button = component.find('.CardListItemEdit').first();
  button.simulate('click');

  setTimeout(() => {
    expect(handleCardEdit.mock.calls.length).toBe(1);
    expect(handleCardEdit.mock.calls[0][0]).toBe(card.card_id);
    done();
  });
});

it('click deleteCard button calls handler', (done) => {
  const handleCardDelete = jest.fn(() => {});
  const component = shallow(<CardListItemComp card={card} onCardDelete={handleCardDelete} />);
  const button = component.find('.CardListItemDelete').first();
  button.simulate('click');

  setTimeout(() => {
    expect(handleCardDelete.mock.calls.length).toBe(1);
    expect(handleCardDelete.mock.calls[0][0]).toBe(card.card_id);
    done();
  });
});

it('click select Card text calls handler', (done) => {
  const handleCardSelect = jest.fn(() => {});
  const component = shallow(<CardListItemComp card={card} onCardSelect={handleCardSelect} />);
  const button = component.find('.CardListItemText').first();
  button.simulate('click');

  setTimeout(() => {
    expect(handleCardSelect.mock.calls.length).toBe(1);
    expect(handleCardSelect.mock.calls[0][0]).toBe(card.card_id);
    done();
  });
});
