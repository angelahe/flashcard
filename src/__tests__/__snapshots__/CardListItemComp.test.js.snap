// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of UI renders consistently 1`] = `
<div
  className="ItemBox AppListItem CardListItemComp"
  role="button"
  tabIndex="0"
>
  <span
    className="CardListItemText DetailText"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="0"
  >
    ABCD-1234
  </span>
  <button
    className="CardListItemEdit AppBtn"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="Add"
      className="btnImg"
      src="edit_FFFFFF.png"
    />
  </button>
  <button
    className="CardListItemDelete AppBtn Delete"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="Delete"
      className="btnImg"
      src="delete_FFFFFF.png"
    />
  </button>
  <hr />
</div>
`;
